packages/near-lake-primitives/src/types/block.ts

```diff
@@ -0,0 +1,156 @@
+import { Action, Receipt } from './receipts';
+import { StreamerMessage, ValidatorStakeView } from './core/types';
+import { Transaction } from './transactions';
+import { Event, RawEvent } from './events';
+import { StateChange } from './stateChanges';
+
+export class Block {
+    private _actions: Map<string, Action>;
+    private _events: Map<string, Event[]>;
+    private _stateChanges: StateChange[];
+
+    constructor(
+        readonly streamerMessage: StreamerMessage,
+        private executedReceipts: Receipt[],
+        readonly postponedReceipts: Receipt[],
+        readonly transactions: Transaction[],
+        actions: Map<string, Action>,
+        events: Map<string, Event[]>,
+        stateChanges: StateChange[]) {
+        this._actions = actions;
+        this._events = events;
+        this._stateChanges = stateChanges;
```

Comment:
```suggestion
        private _actions: Map<string, Action>,
        private _events: Map<string, Event[]>,
        private _stateChanges: StateChange[]) {
```
These can also use parameter properties shorthand

Reply:
They can be but I would have to keep the parameter names with an undescore so I do not have a naming conflict between the getters. I guess that is a fine trade off.